hub/api/hashlist.py

```diff
@@ -0,0 +1,33 @@
+from hub.core.storage.lru_cache import LRUCache
+from typing import Any, Dict, Optional, Union, Sequence
+from hub.core.storage.cachable import CachableCallback, use_callback
+
+
+class Hashlist(CachableCallback):
+    def __init__(self):
+        """Contains list of hashes generated for a particular tensor."""
+        self.hashes = []
+        super().__init__()
+
+    def append(self, hash):
+        self.hashes.append(hash)
+
```

Comment:
im worried about the architecture for this feature, are we sure we really want to be appending a ton of hashes to a single list? 

i did some math, for 100M samples we would have 12.8GB of hashes. we would need to shard this as well?

maybe it would be a good idea just to have another tensor instead of creating a list like this. the only function here really is append.

we need some way to just create a linked tensor instead of another meta file. like when i set hasing to true, whenever i append to one tensor it automatically appends to another linked tensor but with some transformed (hashed) data.

appending to this linked tensor should be locked (not updatable by the user).

it would also be nice to have a system like this that works for other use cases though. like if i want to append all data to a tensor and then have the grayscale version also be implicitly appended, it could be done in the same way.

Reply:
also, a suggestion: stochastic comparisons. for hashing, you may be able to calculate the IOU score with a subset of the data as opposed to the entire sets. not sure how this would work with ordering but if hashes could be sorted you could probably just check the first like N elements in those sorted lists and then the last N and see the score matchup. then you don't need to worry about the middle pieces which could be gigabytes

Reply:
I can add both types of comparison, IOU on subset + entire set. Stochastic comparisons won't work well for cases when a user wants to check if the datasets being compared are different by a few samples. It's better for cases when a user wants to check if the dataset they're uploading is already on Activeloop Cloud (no need to download the entire hashlist).

Reply:
@mccrearyd I wasn't too worried about the scaling as this feature is opt-in and mostly relevant to datasets present on hub cloud. Each hash is 8 bytes, so for a million samples it's only going to be 8MB in size, which isn't too bad even if we fetch the entire hashlist.